contrib/syslog-debun

```diff
@@ -6,6 +7,17 @@
 ### This software may be used and distributed according to the terms of GNU GPLv2
 ### http://www.gnu.org/licenses/gpl-2.0.html
 
+### Check for "local" variable support
+if type local >/dev/null; then
+	:
+elif type bash >/dev/null; then
```

Comment:
Bash is not available oob on all of the syslog-ng's supported platform.
Eg. couple of freebsd's is amongst the supported platform, where bash is not available by default.
But on those systems the /bin/sh proven to be a good posix compatible shell, with local variable support.
On the other hand a couple of Solaris systems were also on the support matrix, where the /bin/sh was also posix compliant, but didn't provide local variables. But /bin/bash is available there, so exec()-ing "ourselves" to bash was a good alternative.
Based on my tests on the supported systems this two case was what I found:
#1 /bin/sh supports local variables and posix compliant
#2 /bin/sh doesn't support local variables, but /bin/bash is available oob. And /bin/bash supports local variables.
Hint: be careful with bash! Some of the supported platforms may contain a very early, like 2.0.something bash version. They follow posix rules, but don't depend on modern bash things!
If you depend only on what posix demands from a shell, then the script can remain as portable, as it was previously.